security/pkg/nodeagent/sds/sdsservice.go

```diff
@@ -168,8 +121,10 @@ func newSDSService(st security.SecretManager, options *security.Options, pkpConf
 	return ret
 }
 
-func (s *sdsservice) generate(resourceNames []string) (model.Resources, error) {
-	resources := model.Resources{}
+var version uberatomic.Uint64
+
+func (s *sdsservice) generate(resourceNames []string) (*discovery.DiscoveryResponse, error) {
```

Comment:
feel a little duplicate if we write a bunch of methods again

Reply:
You mean the version? Yes, but it's just two lines. It would be more code if we share.